nemo_aligner/utils/utils.py

```diff
@@ -469,3 +493,9 @@ def make_sharded_tensors_from_reference(reference_param, model_param, prefix: st
         tuple(model_param.shape) == reference_param.local_shape
     ), f"Model shape ({tuple(model_param.shape)} does not match reference shape ({reference_param.local_shape})"
     return replace(reference_param, key=f"{prefix}.{reference_param.key}", data=model_param, dtype=model_param.dtype)
+
+
+def log_memory(prefix):
```

Comment:
Not used anywhere?
(NB: if we want to use it, we should probably add some `torch.cuda.synchronize()`-- at least optionally -- to get accurate measurements at a specific place in the code)

Reply:
Adding back: sounds good

`torch.cuda.synchronize()`: looks like you're right! I ran some quick toy test and indeed, it can return the proper memory allocation even before it's done running all kernels. And if I try to "trick it" by queueing an op whose output shape depends on input values, it's actually synchronizing automatically in that case (though I didn't try too hard). Interesting, thanks for pointing this out! :)